### ISSUE TITLE
Upgrading ServiceControl to 4.21, 4.21.1, or 4.21.2 can cause deserialization exceptions resulting in metrics/heartbeat/customcheck/saga audit messages being sent to the error queue

### ISSUE THREAD

**Issue by boblangley** [Particular/ServiceControl#2745]
After installing ServiceControl 4.21.1 and upgrading a monitoring instance some customers are reporting metrics message ingestion failures:

```
NServiceBus.MessageDeserializationException: An error occurred while attempting to extract logical messages from incoming physical message 00000000-0000-0000-0000-000000000000 ---> Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: @. Path '', line 0, position 0.
at Newtonsoft.Json.JsonTextReader.ParseValue()
at NServiceBus.Newtonsoft.Json.JsonMessageSerializer.IsArrayStream(Stream stream)
at NServiceBus.Newtonsoft.Json.JsonMessageSerializer.Deserialize(Stream stream, IList`1 messageTypes)
at NServiceBus.DeserializeMessageConnector.Extract(IncomingMessage physicalMessage) in /_/src/NServiceBus.Core/Pipeline/Incoming/DeserializeMessageConnector.cs:line 119
at NServiceBus.DeserializeMessageConnector.ExtractWithExceptionHandling(IncomingMessage message) in /_/src/NServiceBus.Core/Pipeline/Incoming/DeserializeMessageConnector.cs:line 47
--- End of inner exception stack trace ---
at NServiceBus.DeserializeMessageConnector.ExtractWithExceptionHandling(IncomingMessage message) in /_/src/NServiceBus.Core/Pipeline/Incoming/DeserializeMessageConnector.cs:line 51
at NServiceBus.DeserializeMessageConnector.<Invoke>d__1.MoveNext() in /_/src/NServiceBus.Core/Pipeline/Incoming/DeserializeMessageConnector.cs:line 31
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at NServiceBus.ProcessingStatisticsBehavior.<Invoke>d__0.MoveNext() in /_/src/NServiceBus.Core/Performance/Statistics/ProcessingStatisticsBehavior.cs:line 25
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at NServiceBus.TransportReceiveToPhysicalMessageConnector.<Invoke>d__1.MoveNext() in /_/src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveToPhysicalMessageConnector.cs:line 39
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at NServiceBus.RetryAcknowledgementBehavior.<Invoke>d__2.MoveNext() in /_/src/NServiceBus.Core/ServicePlatform/Retries/RetryAcknowledgementBehavior.cs:line 27
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at NServiceBus.MainPipelineExecutor.<Invoke>d__1.MoveNext() in /_/src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs:line 45
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at NServiceBus.Transport.AzureServiceBus.MessagePump.<ReceiveMessage>d__25.MoveNext() in /_/src/Transport/Receiving/MessagePump.cs:line 239
```

**Comment by vw-bnauta** [comment]
We are running into the same issue on 4.21.2 and it's severely affecting our operations.

**Comment by boblangley** [comment]
We are attempting to create a reproduction of this so we can determine the cause. Any additional information you can share will be helpful. We would like to know:

- What version did you upgrade from/to.
- What packages and their versions are you using in your endpoints that are sending metrics.

**Comment by boblangley** [comment]
We [have been able to reproduce this bug](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/issues/454) and are working on releasing a new version of ServiceControl with the fix.

This occurs when a message is sent from an endpoint using the [legacy Azure Service Bus transport](https://docs.particular.net/transports/azure-service-bus/legacy/) and is processed by an endpoint using the [2.0.1 version of the Azure Service Bus transport](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/releases/tag/2.0.1).

**Comment by boblangley** [comment]
We have released ServiceControl version 4.21.3 which contains this fix: https://discuss.particular.net/t/servicecontrol-4-21-3-critical-patch-release-available/2874

Please upgrade your ServiceControl instances to this version if you are affected by this issue.

Also, please be aware that the [legacy Azure Service Bus transport](https://docs.particular.net/transports/azure-service-bus/legacy/) will soon fall out of support. It is advised to upgrade endpoints using the legacy Azure Service Bus transport as soon as possible. Please contact support if you need help doing so.
